src/devices/machine/cat702.h

```diff
@@ -42,4 +46,14 @@ class cat702_device : public device_t
 	devcb_write_line m_dataout_handler;
 };
 
-#endif // MAME_SONY_CAT702_H
+class cat702_piu_device : public cat702_device
+{
+public:
+	cat702_piu_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock)
+		: cat702_device(mconfig, CAT702_PIU, tag, owner, clock) {}
+
+	void write_select(int state);
+	void write_clock(int state);
```

Comment:
Since these are non-virtual and shadow base class members, it creates potential confusion if someone has a `cat702_device &` or `cat702_device *` that refers to an instance of the derived class.  Calling `write_select` or `write_clock` will call the the base class member, not the derived class member.

To avoid confusion, you can either:
* Make a common abstract base class for the two concrete device classes which lacks the `write_select` and `write_clock` member functions so you need a reference to one concrete class or the other.
* Make them `virtual`, which may have a small performance penalty for the indirect call (in practice, if it’s called using a delegate the extra overhead can often be eliminated).